src/app/server/Server.cpp

```diff
@@ -70,6 +73,115 @@ constexpr bool useTestPairing()
+    // It's not an error if the key doesn't exist. Just return right away.
+    VerifyOrReturnError(PersistedStorage::KeyValueStoreMgr().Get(kAdminTableCountKey, &nextAvailableId) == CHIP_NO_ERROR,
+                        CHIP_NO_ERROR);
+    ChipLogProgress(AppServer, "Next available admin ID is %d", nextAvailableId);
+
+    // TODO: The admin ID space allocation should be re-evaluated. With the current approach, the space could be
+    //       exhausted while IDs are still available (e.g. if the admin IDs are allocated and freed over a period of time).
+    //       Also, the current approach can make ID lookup slower as more IDs are allocated and freed.
+    for (AdminId id = 0; id < nextAvailableId; id++)
+    {
+        AdminPairingInfo * admin = adminPairings.AssignAdminId(id);
+        // Recreate the binding if one exists in persistent storage. Else skip to the next ID
+        if (admin->FetchFromKVS(gServerStorage) != CHIP_NO_ERROR)
+        {
+            adminPairings.ReleaseAdminId(id);
+        }
+        else
+        {
+            ChipLogProgress(AppServer, "Found admin pairing for %d, node ID %llu", admin->GetAdminId(), admin->GetNodeId());
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void DeleteAdminPairingsInKVS(AdminId nextAvailableId)
+{
+    PersistedStorage::KeyValueStoreMgr().Delete(kAdminTableCountKey);
+
+    for (AdminId id = 0; id < nextAvailableId; id++)
+    {
+        AdminPairingInfo::DeleteFromKVS(gServerStorage, id);
+    }
+}
+
+static CHIP_ERROR RestoreAllSessionsFromKVS(SecureSessionMgr & sessionMgr, RendezvousServer & server)
+{
+    uint16_t nextSessionKeyId = 0;
+    // It's not an error if the key doesn't exist. Just return right away.
+    VerifyOrReturnError(PersistedStorage::KeyValueStoreMgr().Get(kStorablePeerConnectionCountKey, &nextSessionKeyId) ==
+                            CHIP_NO_ERROR,
+                        CHIP_NO_ERROR);
+    ChipLogProgress(AppServer, "Found %d stored connections", nextSessionKeyId);
+
+    for (uint16_t keyId = 0; keyId < nextSessionKeyId; keyId++)
+    {
+        StorablePeerConnection connection;
+        if (CHIP_NO_ERROR == connection.FetchFromKVS(gServerStorage, keyId))
+        {
+            // The following is static to save on stack space usage
+            static PASESession session;
+            connection.GetPASESession(session);
+
+            ChipLogProgress(AppServer, "Fetched the session information: from %llu", session.PeerConnection().GetPeerNodeId());
+            sessionMgr.NewPairing(Optional<Transport::PeerAddress>::Value(session.PeerConnection().GetPeerAddress()),
+                                  session.PeerConnection().GetPeerNodeId(), &session,
+                                  SecureSessionMgr::PairingDirection::kResponder, connection.GetAdminId(), nullptr);
+        }
+    }
+
+    server.GetRendezvousSession()->SetNextKeyId(nextSessionKeyId);
+    return CHIP_NO_ERROR;
+}
+
+void DeleteSessionsInKVS(uint16_t nextSessionKeyId)
```

Comment:
I would use EraseAllUpTo or ClearAllUpTo to make it obvious we do not reference a single ID but loop through them.